Source/WebCore/platform/ScrollableArea.cpp

```diff
@@ -176,12 +176,23 @@ void ScrollableArea::scrollToPositionWithAnimation(const FloatPoint& position, c
 {
     LOG_WITH_STREAM(Scrolling, stream << "ScrollableArea " << this << " scrollToPositionWithAnimation " << position);
 
+    if (position == scrollPosition())
```

Comment:
I think this early return needs to be after the call to `scrollAnimator().cancelAnimations()`. There might be an enqueued scroll animation that takes us away from the current scroll position.

Reply:
Done